### PR TITLE
Why Haskell post 

### PR DESCRIPTION
**Pull request by arbus**

### CODE REVIEW

File: blog/2019-03-29-why-haskell.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+---
+title: Haskell - The Good, the Bad, The Ugly
+description: A short summary of Holmusk's experience with Haskell so far
+tags: haskell
+---
```
Comment (vrom911) [inline blog/2019-03-29-why-haskell.md:5]: Should we add `author` context?

File: blog/2019-03-29-why-haskell.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+### Cost efficiency
+
+> If you need to scale, Haskell’s runtime efficiency can save you a substantial amount of money, especially at the beginning when you are the most resource constrained.
+
+The Haskell backend that replaced our old backend was significantly more efficient. It allowed us to run fewer servers to support the same workload. The cost savings from a smaller AWS bill can make a difference if your startup is in the phase where every dollar counts.
+
+Here is a chart of our AWS expenditure as we switched to a Haskell backend:
+
+<Insert chart>
```
Comment (vrom911) [inline blog/2019-03-29-why-haskell.md]: TODO

File: blog/2019-03-29-why-haskell.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+Apart from things like `Yesod`, there aren’t really many framework style Haskell projects that come with best practices on how to structure your application or guides on how to do common tasks like handling file uploads, user authentication, database management etc.
+
+Working on Haskell projects feels very much like working with `react` projects in that you have to bring in external libraries piecemeal for most of the features you want and the libraries aren’t necessarily designed with the assumption that they will be used together in that particular combination.
+
+Conventions also codify some hard-learned lessons, in their absence most of those lessons have to be re-learned by your team.
+
+### Lack of libraries
+
+> While the quality of libraries is excellent, the quantity can be rather limiting. Don’t expect the equivalent of passportjs or paperclip.
```
Comment (vrom911) [inline blog/2019-03-29-why-haskell.md]: Do you want to make all libraries/frameworks/tools mentioned in the post as links? That could be handy I guess

File: blog/2019-03-29-why-haskell.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+## The Ugly
+
+### Breaking changes in libraries
+
+> You can expect libraries to have breaking changes that have ripple effects throughout your project.
+
+Most Haskell projects don’t really have a concept of backwards compatibility. They regularly release breaking changes because they tend to be either hobby projects or projects that simply have ambitious technical goals which don’t necessarily align with your company’s goals of having stable interfaces.
+
+Upgrading to a new stack LTS almost always ends up becoming a blocking task that requires many tangential changes or reviewing changelogs carefully of all the libraries that you depend on to make sure that there aren’t any behaviour changes.
```
Comment (vrom911) [inline blog/2019-03-29-why-haskell.md]: Probably not to make it build-tool opinionated but to say upgrading to the newer GHC or just to the newer versions?

File: blog/2019-03-29-why-haskell.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+### The bait of type safety
+
+> Type-safety can be an alluring thing that you eventually end up spending too much time and effort to achieve. Typing everything has diminishing returns.
+
+Having experienced the bliss of GHC being your programming assistant, it can be extremely tempting to rely on evermore fancy type system features to make more illegal states unrepresentable.
+
+Soon, you find yourself encoding entire chunks of your business logic into the type system while a simple run-time check might have sufficed. Just because Haskell allows you to express something in the type system doesn’t mean that it is a good idea to do so.
```
Comment (vrom911) [inline blog/2019-03-29-why-haskell.md]: that it is __always__ a good idea to do so.

Because many times it is.

**Review by vrom911 — APPROVED** [review]

**Review by chshersh — APPROVED** [review]

**Pull request merged**